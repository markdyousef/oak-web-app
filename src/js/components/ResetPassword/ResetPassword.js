// @flow
import React, { Component } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router';
import Input from '../shared/Input';
import { Box, ErrorMessage, ChangePage } from '../../styles';
import { NextButton } from '../shared/Button';
import Arrow from '../../icons/rightArrow';

const Container = styled.section`
    width: 100%;
`;

type Props = {
    resetPassword: (token: string, password: string) => Object
};

type State = {
    password: string,
    passwordRepeat: string,
    message: ?string,
    token: string
};

type DefaultProps = {};

export default class ResetPassword extends Component<DefaultProps, Props, State> {
    static defaultProps: DefaultProps;
    props: Props;
    state: State;
    constructor() {
        super();
        this.state = {
            password: '',
            passwordRepeat: '',
            message: '',
            token: ''
        };
    }
    componentWillMount() {
        const token = new URLSearchParams(location.search).get('token');
        this.setState({ token });
    }
    onSubmit = () => {
        const { resetPassword } = this.props;
        const { password, passwordRepeat, token } = this.state;
        this.setState({ message: null });

        // password validation
        if (password.length < 6) {
            this.setState({ message: 'Your password must be at least 6 characters' });
            return;
        }
        // check if passwords match
        if (password !== passwordRepeat) {
            this.setState({ message: "Your passwords doesn't match." });
            return;
        }
        resetPassword(token, password)
            .then(res => console.log(res))
            .catch(err => console.log(err));
    }
    render() {
        const { password, passwordRepeat, message } = this.state;
        return (
            <Container>
                <Box>
                    <h1>Reset Password</h1>
                    <div>
                        <Input
                            title="new password"
                            value={password}
                            onChange={value => this.setState({ password: value, message: null })}
                            type="password"
                            placeholder="Enter a new password"
                        />
                        <Input
                            title="confirm password"
                            value={passwordRepeat}
                            onChange={value => this.setState({ passwordRepeat: value, message: null })}
                            type="password"
                            placeholder="Confirm new password"
                            onKeyDown={(e) => {
                                if (e.keyCode === 13) {
                                    this.onSubmit();
                                }
                            }}
                        />
                    </div>
                    <div>
                        <NextButton
                            onClick={this.onSubmit}
                            icon={<Arrow />}
                            text="Reset Password"
                        />
                    </div>
                    <div>
                        {message &&
                            <ErrorMessage>
                                <h5>{message}</h5>
                            </ErrorMessage>
                        }
                    </div>
                </Box>
                <ChangePage>
                    <p>Did you remember your old password? <Link to="/login">Sign in!</Link></p>
                </ChangePage>
            </Container>
        );
    }
}
