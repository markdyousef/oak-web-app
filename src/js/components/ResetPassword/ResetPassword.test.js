// @flow
import React from 'react';
import { shallow } from 'enzyme';
import ResetPassword from './ResetPassword';

it('renders correctly', () => {
    const props = {};

    const shallowComponent = shallow(
        <ResetPassword {...props} />
    );
    expect(shallowComponent).toMatchSnapshot();
});
