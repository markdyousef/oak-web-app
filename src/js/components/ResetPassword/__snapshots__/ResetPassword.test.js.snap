// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.section>
    <styled.div>
        <h1>
            Reset Password
        </h1>
        <div>
            <Input
                notValid={false}
                onChange={[Function]}
                onKeyDown={null}
                placeholder="Enter a new password"
                title="new password"
                type="password"
                value=""
            />
            <Input
                notValid={false}
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Confirm new password"
                title="confirm password"
                type="password"
                value=""
            />
        </div>
        <div>
            <Unknown
                icon={<Unknown />}
                onClick={[Function]}
                text="Reset Password"
            />
        </div>
        <div />
    </styled.div>
    <styled.div>
        <p>
            Did you remember your old password? 
            <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/login"
            >
                Sign in!
            </Link>
        </p>
    </styled.div>
</styled.section>,
  "nodes": Array [
    <styled.section>
      <styled.div>
            <h1>
                  Reset Password
            </h1>
            <div>
                  <Input
                        notValid={false}
                        onChange={[Function]}
                        onKeyDown={null}
                        placeholder="Enter a new password"
                        title="new password"
                        type="password"
                        value=""
                  />
                  <Input
                        notValid={false}
                        onChange={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Confirm new password"
                        title="confirm password"
                        type="password"
                        value=""
                  />
            </div>
            <div>
                  <Unknown
                        icon={<Unknown />}
                        onClick={[Function]}
                        text="Reset Password"
                  />
            </div>
            <div />
      </styled.div>
      <styled.div>
            <p>
                  Did you remember your old password? 
                  <Link
                        onlyActiveOnIndex={false}
                        style={Object {}}
                        to="/login"
                  >
                        Sign in!
                  </Link>
            </p>
      </styled.div>
</styled.section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ResetPassword
        params={Object {}}
        resetPassword={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ResetPassword {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onSubmit": [Function],
        "props": Object {
          "params": Object {},
          "resetPassword": [Function],
        },
        "refs": Object {},
        "state": Object {
          "message": "",
          "password": "",
          "passwordRepeat": "",
          "token": undefined,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.section>
          <styled.div>
                    <h1>
                              Reset Password
                    </h1>
                    <div>
                              <Input
                                        notValid={false}
                                        onChange={[Function]}
                                        onKeyDown={null}
                                        placeholder="Enter a new password"
                                        title="new password"
                                        type="password"
                                        value=""
                              />
                              <Input
                                        notValid={false}
                                        onChange={[Function]}
                                        onKeyDown={[Function]}
                                        placeholder="Confirm new password"
                                        title="confirm password"
                                        type="password"
                                        value=""
                              />
                    </div>
                    <div>
                              <Unknown
                                        icon={<Unknown />}
                                        onClick={[Function]}
                                        text="Reset Password"
                              />
                    </div>
                    <div />
          </styled.div>
          <styled.div>
                    <p>
                              Did you remember your old password? 
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/login"
                              >
                                        Sign in!
                              </Link>
                    </p>
          </styled.div>
</styled.section>,
        "_debugID": 2,
        "_renderedOutput": <styled.section>
          <styled.div>
                    <h1>
                              Reset Password
                    </h1>
                    <div>
                              <Input
                                        notValid={false}
                                        onChange={[Function]}
                                        onKeyDown={null}
                                        placeholder="Enter a new password"
                                        title="new password"
                                        type="password"
                                        value=""
                              />
                              <Input
                                        notValid={false}
                                        onChange={[Function]}
                                        onKeyDown={[Function]}
                                        placeholder="Confirm new password"
                                        title="confirm password"
                                        type="password"
                                        value=""
                              />
                    </div>
                    <div>
                              <Unknown
                                        icon={<Unknown />}
                                        onClick={[Function]}
                                        text="Reset Password"
                              />
                    </div>
                    <div />
          </styled.div>
          <styled.div>
                    <p>
                              Did you remember your old password? 
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/login"
                              >
                                        Sign in!
                              </Link>
                    </p>
          </styled.div>
</styled.section>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ResetPassword
    params={Object {}}
    resetPassword={[Function]}
/>,
}
`;
